### DISCUSSION TITLE
Welcome to mkconnect-python Discussions!

### DISCUSSION THREAD

**Discussion by J0EK3R** [J0EK3R/mkconnect-python#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by B311s** [comment]
Hello,
I find the project very interesting, I built the robot dog from MK 15066 and I would like to operate it with a raspi or arduino. The motor + controller is different from the typical units, it comprises two perpendicular motors and battery in one box. Did you try to connect to this aswell? So far I did not find the BLE-signal from it, I tried with an arduino and a raspi. (Included remote control works)

**Comment by J0EK3R** [comment]
Hi @B311s 
Wich Mould King motor device is used in your model? MK3.8?

You could try to change the byte array to the one for the MK3.8:
In file https://github.com/J0EK3R/mkconnect-python/blob/main/MouldKing/MouldKing_Hubs_4_12Ch.py
in line 19 and 21 two byte arrays are defined:

```
    __telegram_connect = bytes([0xAD, 0x7B, 0xA7, 0x80, 0x80, 0x80, 0x4F, 0x52]) # Define a byte array for Telegram Connect

    __telegram_base = bytes([0x7D, 0x7B, 0xA7, 0x88, 0x88, 0x88, 0x88, 0x88, 0x88, 0x82]) # byte array for base Telegram
```

You could try to change them to

```
    __telegram_connect = bytes([0xB1, 0x7B, 0xA7, 0x80, 0x80, 0x80, 0x4F, 0xC1]) # Define a byte array for Telegram Connect

    __telegram_base = bytes([0x81, 0x7B, 0x00, 0x99, 0x99, 0x99, 0x99, 0x99, 0x99, 0xC2]) # byte array for base Telegram

```
and try to connect with Module4_0.Device0...

Here is all I know about MK devices:
https://github.com/vicocz/brickcontroller2/tree/default/BrickController2/BrickController2/DeviceManagement/MouldKing

---

Ahhh, good idea - install the BrickController2 App from vicocz repo - there I've implemented MK4.0, MK6.0, MK3.8 and MK5.0.
If you are able to control your device with the App, we're on the right way:
https://github.com/vicocz/brickcontroller2

**Reply by B311s** [reply]
Thank you very much for the suggestions but the BrickController2 does not connect to the MK device from the 15066.

**Reply by J0EK3R** [reply]
I need to know wich MK device is in your model.
What does the MK app say - wich device do you choose in the app?

---

...and you've shurely installed the BrickController2 app from vicocz - not from the appstore?
And you've pressed the add button not the search button in thr app and tried all MK devices?

![Screenshot_2026-01-26-18-57-31-300_cz vico brickcontrollerlegacy](https://github.com/user-attachments/assets/ad662ea8-a61f-485c-9573-1677fcde4149)

**Reply by B311s** [reply]
Yes sorry I did not download the version from vicocz, I tried from App store. But thank you for the screenshot, it is a MK 3.8 module.

**Reply by B311s** [reply]
As far as I have seen MK 3.8 does not communicate via BLE but 2.4GHz, is that correct?

**Reply by J0EK3R** [reply]
> Yes sorry I did not download the version from vicocz, I tried from App store. But thank you for the screenshot, it is a MK 3.8 module.

Here we go, we're on the right way! ;)

> As far as I have seen MK 3.8 does not communicate via BLE but 2.4GHz, is that correct?

No. MK 3.8 should work.
Did you manage to control your device via vicocz' app?
If not, turn off the MK Controller - when switched on it may turn your device into 2.4GHz mode.
Switch your device off and on. Now the app should be able to control your device.
Another way is the MK app - the app should be able to control your device, right?

And your device is the MK 3.8?
